api/src/services/users.ts

```diff
@@ -95,114 +98,109 @@ export class UsersService extends ItemsService {
-		const emails = data
-			.map((payload: Record<string, any>) => payload.email)
-			.filter((e) => e)
-			.map((e) => e.toLowerCase()) as string[];
+		const emails = data.map((payload) => payload.email).filter((email) => email);
+		const passwords = data.map((payload) => payload.password).filter((password) => password);
 
-		await this.checkUniqueEmails(emails);
-
-		const passwords = data.map((payload) => payload.password).filter((pw) => pw);
+		if (emails.length) {
+			await this.checkUniqueEmails(emails);
+		}
 
-		if (passwords.length > 0) {
+		if (passwords.length) {
 			await this.checkPasswordPolicy(passwords);
 		}
 
-		return await this.service.createMany(data, opts);
-	}
-
-	async updateOne(key: PrimaryKey, data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey> {
-		const email = data.email?.toLowerCase();
-
-		if (email) {
-			await this.checkUniqueEmails([email], key);
-		}
+		/**
+		 * Sync with auth provider
+		 */
+		await Promise.all(
+			data.map(async (user) => {
+				const provider = auth.getProvider(user.provider ?? DEFAULT_AUTH_PROVIDER);
+				await provider.createUser(user);
+			})
+		);
 
-		if (data.password) {
-			await this.checkPasswordPolicy([data.password]);
-		}
+		return await super.createMany(data, opts);
+	}
 
-		if ('tfa_secret' in data) {
-			throw new InvalidPayloadException(`You can't change the "tfa_secret" value manually.`);
-		}
+	/**
+	 * Update many users by query
+	 */
+	async updateByQuery(query: Query, data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey[]> {
+		const keys = await this.getKeysByQuery(query);
+		return keys.length ? await this.updateMany(keys, data, opts) : [];
+	}
 
-		return await this.service.updateOne(key, data, opts);
+	/**
+	 * Update a single user by primary key
+	 */
+	async updateOne(key: PrimaryKey, data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey> {
+		await this.updateMany([key], data, opts);
+		return key;
 	}
 
+	/**
+	 * Update many users by primary key
+	 */
 	async updateMany(keys: PrimaryKey[], data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey[]> {
-		const email = data.email?.toLowerCase();
-
-		if (email) {
-			await this.checkUniqueEmails([email]);
+		if (data.email) {
+			await this.checkUniqueEmails([data.email]);
 		}
 
 		if (data.password) {
 			await this.checkPasswordPolicy([data.password]);
 		}
 
-		if ('tfa_secret' in data) {
+		if (data.tfa_secret !== undefined) {
 			throw new InvalidPayloadException(`You can't change the "tfa_secret" value manually.`);
 		}
 
-		return await this.service.updateMany(keys, data, opts);
-	}
-
-	async updateByQuery(query: Query, data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey[]> {
-		const email = data.email?.toLowerCase();
-
-		if (email) {
-			await this.checkUniqueEmails([email]);
-		}
-
-		if (data.password) {
-			await this.checkPasswordPolicy([data.password]);
+		if (data.provider !== undefined) {
```

Comment:
We prevent changing the provider after a user is created to prevent future headaches with misconfigured users.